azure/defaults.go

```diff
@@ -329,6 +338,47 @@ func UserAgent() string {
 	return fmt.Sprintf("cluster-api-provider-azure/%s", version.Get().String())
 }
 
+// ARMClientOptions returns default ARM client options for CAPZ SDK v2 requests.
+func ARMClientOptions(resourceManagerEndpoint string) *arm.ClientOptions {
+	opts := &arm.ClientOptions{}
+	switch resourceManagerEndpoint {
+	case PublicCloudMgmtEndpoint:
```

Comment:
my comment became outdated with the last push but re-adding here: we could use `auth.CloudEnvironment()` here to determine the cloud, it would be more reliable and accurate than the base URI as some services are using regional base URIs

Reply:
Doesn't BaseURI (which just returns autorest's Environment.ResourceManagerEndpoint) also keep us using go-autorest? What about AzureCluster.Spec.AzureEnvironment? could we use that?

Reply:
I do think we could use `AzureCluster.Spec.AzureEnvironment`. I took a look at doing this but it didn't seem straightforward to add an `AzureCluster` into the Machine and MachinePool scopes. It does look like the implementation of `authorizer.CloudEnvironment()` could be changed in one place, so I think it's possible going forward, but IDK if it should hold up this PR. It doesn't make much difference until we get down to removing the last bits of `go-autorest`, which will take a while.

Reply:
MachinePoolScope and MachineScope both have `ClusterScoper`, we could add a`AzureEnvironment()` function to it that returns the cloud name